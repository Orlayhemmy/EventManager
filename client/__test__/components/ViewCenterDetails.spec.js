import React from 'react';
import Details from '../../components/CenterDetails/Template/Content/ViewCenterDetails';

describe('View center details Component', () => {
  const onApprove = jest.fn();
  const showHiddenDiv = jest.fn();

  const centerState = {
    centerName: '',
    location: '',
    facilities: '',
    description: '',
    imageUrl: '',
    capacity: '',
    cost: ''
  };
  const wrapper = shallow(<Details
        showHiddenDiv={showHiddenDiv}
        onApprove={onApprove}
        centerState={centerState}
      />)
    ;
  it('should render component without error', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('check the length of the containing div in the component', () => {
    expect(wrapper.find('#centerDetails').length).toEqual(1);
  });
});
