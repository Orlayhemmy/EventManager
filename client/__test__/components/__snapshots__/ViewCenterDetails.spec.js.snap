// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View center details Component should render component without error 1`] = `
<div
  id="centerDetails"
>
  <div
    className="imageUpload"
  >
    <img
      className="img-fluid dropzone"
      src=""
    />
  </div>
  <div
    className="media-body text-center mb-4 mt-4"
  >
    <strong
      className="logo text-primary mb-2"
    />
    <h3
      className="mt-2"
    >
      location
    </h3>
    <p />
    <h3
      className="mt-2"
    >
      capacity
    </h3>
    <p />
    <h3
      className="mt-2"
    >
      facilities
    </h3>
    <p />
    <h3
      className="mt-2"
    >
      price
    </h3>
    <p />
    <h3
      className="mt-2"
    >
      description
    </h3>
    <p />
  </div>
  ...
   
  <i
    className="fa fa-pencil-alt main-color"
    data-toggle-id="editCenterDetails"
    onClick={[MockFunction]}
  >
    edit
  </i>
</div>
`;
